resolwe_bio/processes/expression/slamdunk.py

```diff
@@ -70,37 +82,61 @@ def run(self, inputs, outputs):
         name = basename[:-9]
 
         reads_paths = [reads.path for reads in inputs.reads.fastq] + [reads.path for reads in inputs.reads.fastq2]
-        (Cmd['ln']['-s', inputs.transcriptome.fasta.path, 'transcriptome.fasta'])()
+        (Cmd['ln']['-s', inputs.ref_seq.fasta.path, 'ref_seq.fasta'])()
         concatenated_name = f'{name}_concatenated.fastq.gz'
         (Cmd['cat'][reads_paths] > concatenated_name)()
 
         args = [
-            '-r', 'transcriptome.fasta',
+            '-r', 'ref_seq.fasta',
             '-b', inputs.regions.bed.path,
             '-o', 'slamdunk_all',
             '-n', inputs.max_alignments,
-            '-t', '8',
-            '-rl', inputs.read_length
+            '-t', self.requirements.resources.cores,
+            '-rl', inputs.read_length,
         ]
 
         if inputs.filter_multimappers:
             args.extend(['-m', '-fb', inputs.regions.bed.path])
 
         args.append(concatenated_name)
 
-        (Cmd['slamdunk']['all'][args])()
+        return_code, _, _ = Cmd['slamdunk']['all'][args] & TEE(retcode=None)
+        if return_code:
+            self.error('Slamdunk analysis failed.')
+
+        bam_path = os.path.join('slamdunk_all', 'filter', '*.bam')
+        bam_file = glob.glob(bam_path)[0]
+        if os.path.isfile(bam_file):
+            bam_name = os.path.basename(bam_file)[:-4]
+            (Cmd['samtools']['flagstat', bam_file] > bam_name + '_stats.txt')()
+            Cmd['bamtobigwig.sh']([bam_file, inputs.regions.species, self.requirements.resources.cores])
+            outputs.bam = bam_file
+            outputs.bai = bam_file + '.bai'
+            outputs.stats = bam_name + '_stats.txt'
+            outputs.bigwig = bam_name + '.bw'
+        else:
+            self.error('Slamdunk failed to create a BAM file. Exiting.')
+
+        tcount_path = os.path.join('slamdunk_all', 'count', '*_tcount.tsv')
+        tcount_file = glob.glob(tcount_path)[0]
+        if os.path.isfile(tcount_file):
+            outputs.tcount = tcount_file
+        else:
+            self.error('Slamdunk failed to create a tcount file. Exiting.')
```

Comment:
Can slamdunk finish creating a bam file but not tcount.tsv?

Reply:
It is possible as slamdunk consists of four analyses which are independent of each other. On such case occurs when the specified read length is incorrect which fails the last step when tcount is created.